GOVERNANCE.md

```diff
@@ -164,7 +164,6 @@ reported to the TSC immediately so remedial action can be taken.**
 
 For releasing on LTS release lines, releasers must:
 
-* Be added as members of the GitHub [LTS team](https://github.com/orgs/nodejs/teams/lts)
```

Comment:
This change might require TSC approval since it changes Governance.md

Reply:
I wonder if this dates back to when the group was the Long Term Support (LTS) group and this repository was called "lts" and before it was expanded to include current releases?